src/System.Text.Encodings.Web/tests/JavaScriptStringEncoderTests.cs

```diff
@@ -18,11 +19,29 @@ public partial class JavaScriptStringEncoderTests
         public void TestSurrogate()
         {
             Assert.Equal("\\uD83D\\uDCA9", System.Text.Encodings.Web.JavaScriptEncoder.Default.Encode("\U0001f4a9"));
+
             using (var writer = new StringWriter())
             {
                 System.Text.Encodings.Web.JavaScriptEncoder.Default.Encode(writer, "\U0001f4a9");
                 Assert.Equal("\\uD83D\\uDCA9", writer.GetStringBuilder().ToString());
             }
+
+            Span<char> destination = new char[12];
+            OperationStatus status = System.Text.Encodings.Web.JavaScriptEncoder.Default.Encode("\U0001f4a9".AsSpan(), destination, out int charsConsumed, out int charsWritten, isFinalBlock: true);
+            Assert.Equal(OperationStatus.Done, status);
+            Assert.Equal(2, charsConsumed);
+            Assert.Equal(12, charsWritten);
+            Assert.Equal("\\uD83D\\uDCA9", new string(destination.Slice(0, charsWritten).ToArray()));
+        }
+
+        [Fact]
+        public void TestSurrogateBufferTooSmall()
```

Comment:
How about empty input/output spans?

**These cases:**
empty input, empty output
empty input, non-empty output
non-empty input, empty output

Also, maybe a test that verifies that original contents of the destination are replaced (and only the contents in the expected range, with contents outside of the range remain in-tact).

Reply:
Thanks - will do.